test/perf/perf.psm1

```diff
@@ -0,0 +1,145 @@
+
+    .PARAMETER Filter
+    One or more wildcard patterns to filter the benchmarks to be executed or to be listed.
+
+    .PARAMETER Artifacts
+    Path to the folder where you want to store the artifacts produced from running benchmarks.
+
+    .PARAMETER KeepFiles
+    Indicates to keep all temporary files produced for running benchmarks.
+    #>
+    [CmdletBinding()]
+    param(
+        [ValidatePattern(
+            '^7\.(0|1|2)\.\d+(-preview\.\d{1,2})?$',
+            ErrorMessage = 'The package version is invalid or not supported')]
+        [string] $TargetPSVersion,
+
+        [ValidateSet('flat', 'tree')]
+        [string] $List,
+
+        [string[]] $Filter = '*',
+        [string] $Artifacts,
+        [switch] $KeepFiles
+    )
+
+    Begin {
+        Find-Dotnet
+
+        if ($Artifacts) {
+            $Artifacts = $PSCmdlet.SessionState.Path.GetUnresolvedProviderPathFromPSPath($Artifacts)
+        } else {
+            $Artifacts = Join-Path $PSScriptRoot 'BenchmarkDotNet.Artifacts'
+        }
+
+        if (Test-Path -Path $Artifacts) {
+            Remove-Item -Path $Artifacts -Recurse -Force -ErrorAction Stop
+        }
+    }
+
+    End {
+        try {
+            Push-Location -Path "$PSScriptRoot/benchmarks"
+            $savedOFS = $OFS; $OFS = $null
+
+            if ($TargetPSVersion) {
+                Write-Log -message "Run benchmarks targeting the 'Microsoft.PowerShell.SDK' version $TargetPSVersion"
+                $env:PERF_TARGET_VERSION = $TargetPSVersion
+            } else {
+                Write-Log -message "Run benchmarks targeting the current PowerShell code base"
+            }
+
+            $runArgs = @()
+            if ($List) { $runArgs += '--list', $List }
+            if ($KeepFiles) { $runArgs += "--keepFiles" }
+
+            dotnet run -c release --filter $Filter --artifacts $Artifacts $runArgs
+
+            if (Test-Path $Artifacts) {
+                Write-Log -message "`nBenchmark artifacts can be found at $Artifacts"
+            }
+        }
+        finally {
+            $OFS = $savedOFS
+            $env:PERF_TARGET_VERSION = $null
+            Pop-Location
+        }
+    }
+}
+
+function Compare-BenchmarkResult
+{
+    <#
+    .SYNOPSIS
+    Compare two benchmark run results to find possible regressions.
+
+    .PARAMETER BaseResultPath
+    Path to the benchmark result used as baseline.
+
+    .PARAMETER DiffResultPath
+    Path to the benchmark result to be compared with the baseline.
+
+    .PARAMETER Threshold
+    Threshold for Statistical Test. Examples: 5%, 10ms, 100ns, 1s
+
+    .PARAMETER Noise
+    Noise threshold for Statistical Test.
+    The difference for 1.0ns and 1.1ns is 10%, but it's really just noise. Examples: 0.5ns 1ns.
+    The default value is 0.3ns.
+
+    .PARAMETER Top
+    Filter the diff to top `N` results
+    #>
+    param(
+        [Parameter(Mandatory)]
+        [string] $BaseResultPath,
```

Comment:
Maybe BaselinePath ?

Reply:
I like the `BaseResultPath` better. Let's keep it as is for now.